api_definition_test.go

```diff
@@ -391,3 +393,77 @@ func TestBlacklistLinksMulti(t *testing.T) {
 		t.Error(status)
 	}
 }
+
+func startRPCMock(dispatcher *gorpc.Dispatcher) *gorpc.Server {
+	config.SlaveOptions.UseRPC = true
+	config.SlaveOptions.RPCKey = "test_org"
+	config.SlaveOptions.APIKey = "test"
+
+	server := gorpc.NewTCPServer(":9090", dispatcher.NewHandlerFunc())
+	go server.Serve()
+	config.SlaveOptions.ConnectionString = server.Addr
+
+	RPCCLientSingleton = gorpc.NewTCPClient(server.Addr)
+	RPCCLientSingleton.Conns = 1
+	RPCCLientSingleton.Start()
+	RPCClientIsConnected = true
+	RPCFuncClientSingleton = GetDispatcher().NewFuncClient(RPCCLientSingleton)
+
+	return server
+}
+
+func stopRPCMock(server *gorpc.Server) {
+	config.SlaveOptions.ConnectionString = ""
+	config.SlaveOptions.RPCKey = ""
+	config.SlaveOptions.APIKey = ""
+	config.SlaveOptions.UseRPC = false
+
+	server.Listener.Close()
+	server.Stop()
+
+	RPCCLientSingleton.Stop()
+	RPCClientIsConnected = false
+	RPCClients = map[string]chan int{}
+	RPCCLientSingleton = nil
+	RPCFuncClientSingleton = nil
+}
+
+func TestGetAPISpecsRPCFailure(t *testing.T) {
+	// Mock RPC
+	dispatcher := gorpc.NewDispatcher()
+	dispatcher.AddFunc("GetApiDefinitions", func(clientAddr string, dr *DefRequest) (string, error) {
+		return "malformed json", nil
+	})
+	dispatcher.AddFunc("Login", func(clientAddr string, userKey string) bool {
+		return true
+	})
+
+	rpc := startRPCMock(dispatcher)
+
+	specs := getAPISpecs()
+	if len(specs) != 0 {
+		t.Error("Should return empty value for malformed rpc response", specs)
+	}
+
+	stopRPCMock(rpc)
```

Comment:
have these be a `defer` right after start, for readability and consistency (it will survive if a test exits early).